subprojects/android-lib/snackbar-proxy/src/main/kotlin/com/avito/android/snackbar/proxy/SnackbarProxy.kt

```diff
@@ -0,0 +1,20 @@
+package com.avito.android.snackbar.proxy
+
+import androidx.annotation.VisibleForTesting
+import com.google.android.material.snackbar.Snackbar
+
+// VisibleForTesting doesn't work in Kotlin https://issuetracker.google.com/issues/140642032
+// Fixed in AS 4.0
+@VisibleForTesting
+object SnackbarProxyHolder {
+    var proxy: SnackbarProxy? = null
```

Comment:
I consider don't create the platform realization like in the ToastProxy behaviour

Reply:
Why safe call in `showSnackbar` then? It make sense to fail there

Reply:
When the proxy is null then nothing to do, when the proxy is not null I call a method on it

Reply:
i can't see how production code will use `showSnackbar` method, but i'm afraid that it won't show snackbar when it's needed.
Even if it's not the case, this code style creates that type of anxiety